Xamarin.Essentials/Calendar/Calendar.ios.cs

```diff
@@ -27,16 +27,48 @@ static async Task<IReadOnlyList<ICalendar>> PlatformGetCalendarsAsync()
             return calendarList.AsReadOnly();
         }
 
-        static Task<IReadOnlyList<IEvent>> PlatformGetEventsAsync(string calendarId = null, DateTimeOffset? startDate = null, DateTimeOffset? endDate = null) => throw new NotImplementedException();
-
-        static async Task PlatformRequestCalendarReadAccess()
+        static async Task<IReadOnlyList<IEvent>> PlatformGetEventsAsync(string calendarId = null, DateTimeOffset? startDate = null, DateTimeOffset? endDate = null)
         {
             await Permissions.RequireAsync(PermissionType.CalendarRead);
-        }
 
-        static async Task PlatformRequestCalendarWriteAccess()
-        {
-            await Permissions.RequireAsync(PermissionType.CalendarWrite);
+            var systemAbsoluteReferenceDate = new DateTime(2001, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
+            var eventList = new List<Event>();
+            var startDateToConvert = startDate ?? DateTimeOffset.Now;
+            var endDateToConvert = endDate ?? startDateToConvert.AddYears(4);   // 4 years is the maximum period that a iOS calendar events can search
+            var sDate = NSDate.FromTimeIntervalSinceReferenceDate((startDateToConvert.UtcDateTime - systemAbsoluteReferenceDate).TotalSeconds);
+            var eDate = NSDate.FromTimeIntervalSinceReferenceDate((endDateToConvert.UtcDateTime - systemAbsoluteReferenceDate).TotalSeconds);
+            var calendars = !string.IsNullOrEmpty(calendarId) ? CalendarRequest.Instance.Calendars.Where(x => x.CalendarIdentifier == calendarId).ToArray() : null;
+            var query = CalendarRequest.Instance.PredicateForEvents(sDate, eDate, calendars);
+            var events = CalendarRequest.Instance.EventsMatching(query);
+
+            foreach (var e in events)
+            {
+                eventList.Add(new Event
+                {
+                    Id = e.CalendarItemIdentifier,
+                    Title = e.Title,
+                    Start = (long)Math.Floor((Math.Abs(NSDate.FromTimeIntervalSince1970(0).SecondsSinceReferenceDate) + e.StartDate.SecondsSinceReferenceDate) * 1000),
+                    End = (long)Math.Floor((Math.Abs(NSDate.FromTimeIntervalSince1970(0).SecondsSinceReferenceDate) + e.EndDate.SecondsSinceReferenceDate) * 1000)
+                });
+            }
+            eventList.Sort((x, y) =>
+            {
+                if (!x.StartDate.HasValue)
+                {
+                    if (!y.EndDate.HasValue)
+                    {
+                        return 0;
+                    }
+                    return -1;
+                }
+                return !y.EndDate.HasValue ? 1 : x.StartDate.Value.CompareTo(y.EndDate.Value);
```

Comment:
Not too fussed about this but I would probably split these into an if statement for the `EndDate.HasValue`, return 1 from there and otherwise I would return the CompareTo value

Reply:
I originally had the above, but code suggested I refactor it this way.

Will revert to if statement if preferred?

Reply:
Up to you. Whatever you think is more readable.